packages/wallet-ui/src/hooks/useHasMetamask.ts

```diff
@@ -36,13 +46,20 @@ export const useHasMetamask = () => {
   };
 };
 
-export const detectMetamask = async () => {
+export const detectMetamask = async (): Promise<EIP6963ProviderDetail | null> => {
+  return new Promise((resolve) => {
+    const handleProvider = (providerDetail: any) => {
+      resolve(providerDetail);
+    };
+    requestProvider(handleProvider);
+    setTimeout(() => resolve(null), 5000); // Timeout to handle no response
+  });
+};
+
+export const detectMetamaskLegacy = async () => {
```

Comment:
what is the reason to keep this method if it is going to not correct if 2 wallets installed